src/app/[locale]/(auth)/dashboard/handlebars-playground/handlebars-playground.tsx

```diff
@@ -0,0 +1,158 @@
+'use client';
+
+import Handlebars from 'handlebars';
+import { useRouter, useSearchParams } from 'next/navigation';
+import React, { useEffect, useState } from 'react';
+
+import { Button } from '@/components/ui/button';
+import { TextArea } from '@/components/ui/text-area';
+import { fetchTemplateById, saveTemplate, updateTemplate } from '@/libs/actions/templates';
+import { TemplateType } from '@/types/Template';
+
+const HandlebarsPlayground = () => {
+  const [template, setTemplate] = useState<string>(''); // Handlebars template
+  const [inputData, setInputData] = useState<string>(''); // JSON data for the template
+  const [output, setOutput] = useState<string>(''); // Rendered HTML
+  const [error, setError] = useState<string | null>(null); // Error messages
+
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const templateId = searchParams.get('templateId'); // Get templateId from query params
+
+  const handleRender = () => {
+    setError(null);
+
+    try {
+      // Compile the Handlebars template
+      const compiledTemplate = Handlebars.compile(template);
+
+      // Parse the JSON input
+      const context = inputData.trim() ? JSON.parse(inputData) : {};
+
+      // Render the template with the context
+      const result = compiledTemplate(context);
+      setOutput(result);
+    } catch (err: any) {
+      setError(`Failed to render template: ${err.message}`);
+    }
+  };
+
+  // Automatically update the preview when template or JSON changes
+  useEffect(() => {
+    if (template || inputData) {
+      handleRender();
+    }
+  }, [template, inputData]);
+
+  useEffect(() => {
+    const loadTemplateData = async () => {
+      if (templateId) {
+        const response = await fetchTemplateById(templateId);
+        if (!response.success) {
+          return;
+        }
+        // Load the template data into the editor
+        setTemplate(response.data?.templateContent as string);
+        setInputData(response.data?.templateSampleData as string);
+      }
+    };
+    loadTemplateData();
+    handleRender();
+  }, [templateId]);
+
+  const handleSave = async () => {
+    if (templateId) {
+      try {
+        const response = await updateTemplate({
```

Comment:
you can have the same function to save template - toggle between insert or update based on the templateid provided